### DISCUSSION TITLE
QUESTION: Calculation of Rollup for Prior Data

### DISCUSSION THREAD

**Discussion by AndyHaas** [jamessimone/apex-rollup#188] [Q&A]
Hey got a question as I am getting mixed results when running this process. When I run for all configurations of CMDT on Opportunity not all of the records get updated. I have to run this several times for the records to get updated. 

Can you explain if the process is only going through a select number of records or is the process trying to go through the entire data collection that is in Salesforce?

Refrence Material
https://github.com/jamessimone/apex-rollup#calculating-rollups-after-install

**Comment by jamessimone** [comment]
@AndyHaas it might be better to create new Discussions, Q & A style, instead of logging new issues (for questions). At the moment, the answer is "it's complicated" - currently, `apex-rollup` fetches all child records in order to run the full recalculation. This is something I'd like to improve upon, especially when you have several CMDT records with filtered Calc Item Where Clauses where the full record set would end up being quite a bit less than, say, all Opportunities in an org, but getting all of that working nicely is a larger undertaking.

It should only take one run to go through your whole data set - depending on the size of it, it could be a large batch job with many different chunks, or a Queueable that runs once or several times. I've seen a few CPU timeout errors from your org - it might be helpful for you to tweak the `Batch Chunk Size` property on the `Org_Default` RollupControl__mdt record down from 2000 if that's the case. This is all theory, of course, without something more substantial to work with, but I would imagine that if some of the batches are timing out due to using the default chunk size of 2000 records per iteration, lowering the batch amount will help avoid timeouts - which should in turn lead to all of the data being written correctly in one go.

**Reply by AndyHaas** [reply]
I've reduced my Batch size and changed a few other numbers around but I am still not getting records to be updated. We are doing opportunity rollups to accounts. Getting a currency number from the oppty and summing to update the account. 

I created a report so I can see what the number should be and I am not even getting close to the correct number. 

Any suggestions?

**Reply by jamessimone** [reply]
I would need to see a log(s) with `Is Rollup Logging Enabled` to assist further. It could just be an issue with the `Calc Item Where Clause`, for example, but it's impossible to say without looking hands-on at the data. Let me know how I can assist you - I would be very happy to get this resolved!

**Comment by AndyHaas** [comment]
Closing due to full recal not working as expected.
